src/aws/region.rs

```diff
@@ -36,9 +32,9 @@ impl RegionOrEndpoint {
         }
     }
 
-    pub fn endpoint(&self) -> crate::Result<Option<Endpoint>> {
-        let uri = self.endpoint.as_deref().map(Uri::from_str).transpose()?;
-        Ok(uri.map(Endpoint::immutable))
+    pub fn endpoint(&self) -> crate::Result<Option<String>> {
+        let endpoint = self.endpoint.clone();
+        Ok(endpoint)
```

Comment:
Doesn't strictly need to still be a `Result` but I figured we'd want to reintroduce some endpoint validation at some point and this minimized unnecessary changes in the codebase.